atlas/AtlasMl/atlasml/ml/MLPipelines/PipelineWorkflows.py

```
@@ -0,0 +1,292 @@
+            and competency association.
+        """
+        embedding_id, embedding = generate_embeddings_local(uuid, text)
+        clusters = self.weaviate_client.get_all_embeddings(CollectionNames.CLUSTERCENTER.value)
+        medoids = np.array([entry["vector"]["default"] for entry in clusters])
+        similarity_scores = np.array([compute_cosine_similarity(embedding, medoid) for medoid in medoids])
+        best_medoid_idx = int(np.argmax(similarity_scores))
+        competency = self.weaviate_client.get_embeddings_by_property(CollectionNames.COMPETENCY.value, "cluster_id",
+                                                                     clusters[best_medoid_idx]["properties"][
+                                                                         "cluster_id"])
+        competency_to_match = None
+        if competency: competency_to_match = competency[0]
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)